pkg/kubelet/kuberuntime/labels.go

```diff
@@ -0,0 +1,264 @@
+		annotations[kubernetesPodTerminationGracePeriodLabel] = strconv.FormatInt(*pod.Spec.TerminationGracePeriodSeconds, 10)
+	}
+
+	return annotations
+}
+
+// newContainerLabels creates container labels from api.Container and api.Pod.
+func newContainerLabels(container *api.Container, pod *api.Pod, restartCount int) map[string]string {
+	labels := map[string]string{}
+	labels[types.KubernetesPodNameLabel] = pod.Name
+	labels[types.KubernetesPodNamespaceLabel] = pod.Namespace
+	labels[types.KubernetesPodUIDLabel] = string(pod.UID)
+	if pod.DeletionGracePeriodSeconds != nil {
+		labels[kubernetesPodDeletionGracePeriodLabel] = strconv.FormatInt(*pod.DeletionGracePeriodSeconds, 10)
+	}
+	if pod.Spec.TerminationGracePeriodSeconds != nil {
+		labels[kubernetesPodTerminationGracePeriodLabel] = strconv.FormatInt(*pod.Spec.TerminationGracePeriodSeconds, 10)
+	}
+
+	labels[types.KubernetesContainerNameLabel] = container.Name
+	labels[kubernetesContainerHashLabel] = strconv.FormatUint(kubecontainer.HashContainer(container), 16)
+	labels[kubernetesContainerRestartCountLabel] = strconv.Itoa(restartCount)
+	labels[kubernetesContainerTerminationMessagePathLabel] = container.TerminationMessagePath
+	if container.Lifecycle != nil && container.Lifecycle.PreStop != nil {
+		// Using json enconding so that the PreStop handler object is readable after writing as a label
+		rawPreStop, err := json.Marshal(container.Lifecycle.PreStop)
+		if err != nil {
+			glog.Errorf("Unable to marshal lifecycle PreStop handler for container %q of pod %q: %v", container.Name, format.Pod(pod), err)
+		} else {
+			labels[kubernetesContainerPreStopHandlerLabel] = string(rawPreStop)
+		}
+	}
+
+	return labels
+}
+
+// getPodSandboxInfoFromLabels gets labeledPodSandboxInfo from labels.
+func getPodSandboxInfoFromLabels(labels map[string]string) *labeledPodSandboxInfo {
+	return &labeledPodSandboxInfo{
+		PodName:      getStringValueFromLabel(labels, types.KubernetesPodNameLabel),
+		PodNamespace: getStringValueFromLabel(labels, types.KubernetesPodNamespaceLabel),
+		PodUID:       kubetypes.UID(getStringValueFromLabel(labels, types.KubernetesPodUIDLabel)),
+	}
+}
+
+// getPodSandboxInfoFromAnnotations gets annotatedPodSandboxInfo from annotations.
+func getPodSandboxInfoFromAnnotations(annotations map[string]string) *annotatedPodSandboxInfo {
+	var err error
+
+	podSandboxInfo := &annotatedPodSandboxInfo{
+		Annotations: annotations,
+	}
+
+	if podSandboxInfo.PodDeletionGracePeriod, err = getInt64PointerFromLabel(annotations, kubernetesPodDeletionGracePeriodLabel); err != nil {
+		glog.Errorf("Unable to get %q from annotations %q: %v", kubernetesPodDeletionGracePeriodLabel, annotations, err)
+	}
+	if podSandboxInfo.PodTerminationGracePeriod, err = getInt64PointerFromLabel(annotations, kubernetesPodTerminationGracePeriodLabel); err != nil {
+		glog.Errorf("Unable to get %q from annotations %q: %v", kubernetesPodTerminationGracePeriodLabel, annotations, err)
+	}
+
+	// Remove grace period
```

Comment:
These two fields are already stored at `podSandboxInfo.PodDeletionGracePeriod` and `podSandboxInfo.PodTerminationGracePeriod`.